tests/cupyx_tests/scipy_tests/sparse_tests/test_linalg.py

```diff
@@ -561,3 +562,190 @@ def test_invalid(self):
+                                        A_array.T.dot(xp.array(self.ys)))
+        cupy.testing.assert_array_equal(A.H.matvec(xp.array(self.ys)),
+                                        A_array.T.conj()
+                                                 .dot(xp.array(self.ys)))
+        cupy.testing.assert_array_equal(A * xp.array(self.xs),
+                                        A_array.dot(xp.array(self.xs)))
+        return (A.matvec(xp.array(self.xs)), A*xp.array(self.xs))
+
+    @testing.numpy_cupy_array_equal(sp_name='sp')
+    def test_matmat(self, xp, sp):
+        x2 = xp.array([[1, 4], [2, 5], [3, 6]])
+        M, A_array = self._outer_cases(xp, sp)
+        A = sp.linalg.aslinearoperator(M)
+        cupy.testing.assert_array_equal(A.matmat(x2), A_array.dot(x2))
+        if xp.array(self.ys).ndim == 2:
+            cupy.testing.assert_array_equal(A.T.matmat(xp.array(self.ys)),
+                                            A_array.T.dot(xp.array(self.ys)))
+        return (A.matmat(x2), A*x2)
+
+    @testing.numpy_cupy_array_equal(sp_name='sp')
+    def test_rmatvec(self, xp, sp):
+        M, A_array = self._outer_cases(xp, sp)
+        A = sp.linalg.aslinearoperator(M)
+        cupy.testing.assert_array_equal(
+            A.rmatvec(xp.array(self.ys)), A_array.T.conj()
+                                                   .dot(xp.array(self.ys)))
+        return A.rmatvec(xp.array(self.ys))
+
+    @testing.numpy_cupy_array_equal(sp_name='sp')
+    def test_rmatmat(self, xp, sp):
+        if xp.array(self.ys).ndim < 2:
+            return 1
+        M, A_array = self._outer_cases(xp, sp)
+        A = sp.linalg.aslinearoperator(M)
+        cupy.testing.assert_array_equal(A.rmatmat(xp.array(self.ys)),
+                                        A_array.T.conj()
+                                        .dot(xp.array(self.ys)))
+        return A.rmatmat(xp.array(self.ys))
+
+    @testing.numpy_cupy_array_equal(sp_name='sp')
+    def test_dot(self, xp, sp):
+        M, A_array = self._outer_cases(xp, sp)
+        A = sp.linalg.aslinearoperator(M)
+        M, N = A.shape
+
+        x0 = xp.array([1, 2, 3])
+        x1 = xp.array([[1], [2], [3]])
+        x2 = xp.array([[1, 4], [2, 5], [3, 6]])
+
+        cupy.testing.assert_array_equal(A.dot(x0), A_array.dot(x0))
+        cupy.testing.assert_array_equal(A.dot(x1), A_array.dot(x1))
+        cupy.testing.assert_array_equal(A.dot(x2), A_array.dot(x2))
+
+        return (A.dot(x0), A.dot(x1), A.dot(x2))
+
+    # generate random matrix
+    def _make_matrix(self, dtype, xp):
+        shape = (self.n, self.n)
+        a = testing.shaped_random(shape, xp, dtype=dtype)
+        mask = testing.shaped_random(shape, xp, dtype=dtype, scale=1)
+        a[mask > self.density] = 0
+        return a
+
+    @testing.for_dtypes('fdFD')
+    @testing.numpy_cupy_allclose(rtol=1e-5, atol=1e-5, sp_name='sp',
+                                 contiguous_check=False)
+    def test_sparse(self, dtype, xp, sp):
```

Comment:
Could you split this method into another class? This test is not needed to be parameterized.

Reply:
will remove the `test_dense` and `test_sparse` as previous changes to other `test_*` have made the former 2 functions redundant